app/controllers/vendors_controller.rb

```diff
@@ -23,9 +26,25 @@ def create
     end
   end
 
+  def update
+    @vendor = Vendor.find(params[:id])
+    if Vendor.approve(@vendor)
+      flash[:notice] = "Vendor Updated"
+      redirect_to vendors_path
+    else
+      flash[:notice] = "Error"
+      render vendors_path
+    end
+  end
```

Comment:
The test for this one is admin_approve_delete_spec.rb in the /spec/features folder!

Also, the edit details functionality should come from vendor owner; we're planning on addressing that next.